architect/state_table_generators.py

```diff
@@ -193,6 +201,7 @@ def generate_sparse_table(self, as_of_dates):
         """
         logging.debug('Generating sparse table using as_of_dates: %s', as_of_dates)
         self._generate_sparse_table(self.sparse_table_query_func(as_of_dates))
+        table_should_have_data(self.sparse_table_name, self.db_engine)
```

Comment:
I don't think the generic 'no data' message works well here. It works better for input tables, because when they see the name of the table it should ring a bell.

The sparse state table, rather, is just internal, so the name itself won't mean anything.

We can still use the underlying logic ( `architect.database_reflection.table_has_data` ) to query it. But the message should have more info about why the table is empty. Something like:

'No entities specified in '{name dense state table here}' have time ranges that encompass any of the experiment's as_of_dates. {i would say print the as_of_dates here, but it's an exception so maybe that's too long} Please check both your temporal configuration and state configuration'

I'm a bit torn on how hard to try to generalize this message to encompass both the events->sparse states and dense_state->sparse states paths. It's just about impossible to trigger this error using the events path unless the events table is completely empty, and in that case it should be caught beforehand. But technically this code hits both so we may want to have two possible exception messages controlled by an if statement.

Reply:
Agreed – (I figured you might say so, but wanted to leave it to discussion here). One sec….

Reply:
Refined error messaging. Let me know how it suits.